### PR TITLE
Add detailed statistics for the subcommand `show`

### PR DESCRIPTION
**Pull request by Halbaroth**
It is useful to know how long solvers took to answer `sat`, `unsat` and `unknown`. In this PR, benchpress outputs more statistics for these answers while using the subcommand `show`.

I also add a `.editorconfig` ;)

### CODE REVIEW

**Comment by Halbaroth** [comment]
I will add an option to toggle detailed statistics. Otherwise the CI cannot succeed.

File: src/core/Test_stat.ml [hunk 1]
```diff
@@ -2,16 +2,65 @@ open Common
 open Test
 module PB = PrintBox
 
+(* Aggregate function computing the mean and the standard deviation
+   of data series using the Welford's algoritm. *)
+module Stats = struct
+  type acc = { n: int; total: float; mean: float * float; s: float * float }
```
Comment (c-cube) [inline src/core/Test_stat.ml:8]: you probably need to install ocamlformat 0.24.1 and format the new code!

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
+root = true
```
Comment (c-cube) [inline .editorconfig:1]: this seems like it doesn't belong in the repo :)

File: src/core/Test_stat.ml [hunk 3]
```diff
@@ -24,10 +73,30 @@ let get_timeout r = r.timeout
...
 
+let pb_detail_stats_color c { n; total; mean; sd } =
+  let open PB in
+  if n = 0 then
+    align ~h:`Left ~v:`Center (int n)
+  else
+    vlist
+      [
+        pb_int_color c n;
+        grid
```
Comment (c-cube) [inline src/core/Test_stat.ml:92]: this is not wrong, but I think `PB.record` could be easier :)

**Review by c-cube — COMMENTED** [review]
Looks pretty good, just needs to pass CI + reformat :)

File: .editorconfig (hunk 2 above)
Comment (Halbaroth) [reply .editorconfig:1]: I'm not sure to understand what's the problem? This clause means the editor has to stop while looking for a `.editorconfig` in parent directories of its working directory:
```
When opening a file, EditorConfig plugins look for a file named .editorconfig in the directory of the opened file and in every parent directory. A search for .editorconfig files will stop if the root filepath is reached or an EditorConfig file with root=true is found.
``` 
(source: https://editorconfig.org/#file-location)

File: src/core/Test_stat.ml (hunk 1 above)
Comment (Halbaroth) [reply src/core/Test_stat.ml:8]: I did with the right version of `ocamlformat`.

**Comment by c-cube** [comment]
Hu, I didn't realize it was that portable, my bad.

**Comment by c-cube** [comment]
No sure what the segfault is about, any idea?

**Comment by Halbaroth** [comment]
I can reproduce the bug on OCaml 4.08. I'll investigate.

**Comment by Halbaroth** [comment]
Bad news. The bug is localized in the bindings library `sqlite3`. It's fixed in its release `5.0.3` which requires at least `OCaml 4.12`:
The issue: https://github.com/mmottl/sqlite3-ocaml/issues/47
The release: https://github.com/mmottl/sqlite3-ocaml/releases/tag/5.0.3

**Comment by c-cube** [comment]
Oh yeah. Ok we can bump the minimal version.

**Comment by Halbaroth** [comment]
Sorry, I thought that you're going to bump the minimal version in a separated PR. I did it in the last commit of this PR. Notice that I modified by myself the opam files as it seems you don't use `dune` to generate them. I'm not sure it's correct :)

**Comment by c-cube** [comment]
Can you edit the github workflow to use 4.12 and 5.0 as OCaml versions?

**Comment by c-cube** [comment]
Thank you! :)

**Pull request merged**